rest/blip_api_test.go

```diff
@@ -4494,3 +4494,82 @@ func TestBlipLegacyAttachDocUpdate(t *testing.T) {
 		require.Equal(t, keyValueErr.DocumentID, v2Key)
 	}
 }
+
+// Tests changes made in CBG-2151 to return errors from sendRevision unless it's a document not found error,
+// in which case a noRev should be sent.
+func TestSendRevisionNoRevHandling(t *testing.T) {
+	if !base.UnitTestUrlIsWalrus() {
+		t.Skip("Skip LeakyBucket test when running in integration")
+	}
+	testCases := []struct {
+		name        string
+		error       error
+		expectNoRev bool
+	}{
+		{
+			name:        "not_found",
+			error:       gocb.ErrDocumentNotFound,
+			expectNoRev: true,
+		},
+		{
+			name:        "overload",
+			error:       gocb.ErrOverload,
+			expectNoRev: false,
+		},
+	}
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			rt := NewRestTester(t, &RestTesterConfig{
+				guestEnabled: true,
+				TestBucket:   base.GetTestBucket(t).LeakyBucketClone(base.LeakyBucketConfig{}),
+			})
+			defer rt.Close()
+
+			leakyBucket, ok := base.AsLeakyBucket(rt.Bucket())
+			require.True(t, ok)
+
+			btc, err := NewBlipTesterClientOptsWithRT(t, rt, nil)
+			require.NoError(t, err)
+			defer btc.Close()
+
+			// Change noRev handler so it's known when a noRev is received
+			recievedNoRevs := make(chan *blip.Message)
+			btc.pullReplication.bt.blipContext.HandlerForProfile[db.MessageNoRev] = func(msg *blip.Message) {
+				fmt.Println("Received noRev")
+				recievedNoRevs <- msg
+			}
+
+			resp := rt.SendAdminRequest(http.MethodPut, "/db/"+test.name, `{"foo":"bar"}`)
+			assertStatus(t, resp, http.StatusCreated)
```

Comment:
out of scope for this review but I'd be tempted to do a `s/assertStatus/requireStatus/` since this statement is always a require

Reply:
`assertStatus` actually does a require in the function (for some reason). This should hopefully be fixed in ticket CBG-1984

Reply:
Thanks for PR. Have updated to use new version `requireStatus`

Reply:
I think it's hard to address general issues as in CBG-1984 but it's easier to fix one offs with a quick sed statement.